OptimizelySDK.Tests/DecisionServiceTest.cs

```diff
@@ -441,5 +449,281 @@ public void TestGetVariationWithBucketingId()
+            var expectedVariation = ProjectConfig.GetVariationFromId("test_experiment_multivariate", "122231");
+            var userAttributes = new UserAttributes();
+
+            DecisionServiceMock.Setup(ds => ds.GetVariation(ProjectConfig.GetExperimentFromKey("test_experiment_multivariate"), "user1", userAttributes)).Returns(expectedVariation);
+
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("multi_variate_feature");
+            var variation = DecisionServiceMock.Object.GetVariationForFeatureExperiment(featureFlag, "user1", userAttributes);
+
+            Assert.AreEqual(expectedVariation, variation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is bucketed into experiment \"test_experiment_multivariate\" of feature \"multi_variate_feature\"."));
+        }
+
+        // Should return the variation the user is bucketed into when the user is bucketed into one of the experiments
+        [Test]
+        public void TestGetVariationForFeatureExperimentGivenMutexGroupAndUserIsBucketed()
+        {
+            var mutexExperiment = ProjectConfig.GetExperimentFromKey("group_experiment_1");
+            var expectedVariation = mutexExperiment.Variations[0];
+            var userAttributes = new UserAttributes();
+
+            DecisionServiceMock.Setup(ds => ds.GetVariation(ProjectConfig.GetExperimentFromKey("group_experiment_1"), "user1", userAttributes)).Returns(expectedVariation);
+
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("boolean_feature");
+            var variation = DecisionServiceMock.Object.GetVariationForFeatureExperiment(featureFlag, "user1", userAttributes);
+
+            Assert.AreEqual(expectedVariation, variation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is bucketed into experiment \"group_experiment_1\" of feature \"boolean_feature\"."));
+        }
+
+        // Should return null and log a message when the user is not bucketed into any of the mutex experiments 
+        [Test]
+        public void TestGetVariationForFeatureExperimentGivenMutexGroupAndUserNotBucketed()
+        {
+            var mutexExperiment = ProjectConfig.GetExperimentFromKey("group_experiment_1");
+            
+            DecisionServiceMock.Setup(ds => ds.GetVariation(It.IsAny<Experiment>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns<Variation>(null);
+
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("boolean_feature");
+            var actualVariation = DecisionServiceMock.Object.GetVariationForFeatureExperiment(featureFlag, "user1", new UserAttributes());
+
+            Assert.IsNull(actualVariation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is not bucketed into any of the experiments on the feature \"boolean_feature\"."));
+        }
+
+        #endregion // GetVariationForFeatureExperiment Tests
+
+        #region GetVariationForFeatureRollout Tests
+
```

Comment:
minor nit:
add test description for returning null when no rollout exists for the feature

Reply:
Description added.